ext/opentelemetry-ext-cloud-trace/src/opentelemetry/ext/cloud_trace/__init__.py

```diff
@@ -0,0 +1,249 @@
+                    "display_name": get_truncatable_str(span.name),
+                    "start_time": start_time,
+                    "end_time": end_time,
+                    "parent_span_id": parent_id,
+                    "attributes": attributes,
+                    "links": _extract_links(span.links),
+                    "status": _extract_status(span.status),
+                    "time_events": _extract_events(span.events),
+                }
+            )
+
+        return cloud_trace_spans
+
+    def shutdown(self):
+        pass
+
+
+def get_time_from_ns(ns):
+    """Given epoch nanoseconds, split into epoch milliseconds and remaining nanoseconds"""
+    if not ns:
+        return None
+    return {"seconds": int(ns / 1e9), "nanos": int(ns % 1e9)}
+
+
+def get_truncatable_str(str_to_convert, max_length=MAX_LENGTH):
+    """Truncate a string if exceed limit and record the truncated bytes
+        count.
+    """
+    truncated, truncated_byte_count = check_str_length(
+        str_to_convert, max_length
+    )
+
+    result = {
+        "value": truncated,
+        "truncated_byte_count": truncated_byte_count,
+    }
+    return result
+
+
+def check_str_length(str_to_check, limit=MAX_LENGTH):
+    """Check the length of a string. If exceeds limit, then truncate it.
+    """
+    str_bytes = str_to_check.encode("utf-8")
+    str_len = len(str_bytes)
+    truncated_byte_count = 0
+
+    if str_len > limit:
+        truncated_byte_count = str_len - limit
+        str_bytes = str_bytes[:limit]
+
+    result = str(str_bytes.decode("utf-8", errors="ignore"))
+
+    return result, truncated_byte_count
+
+
+def _extract_status(status: trace_api.Status):
+    """Convert a Status object to dict."""
+    if not status:
+        return None
+    status_json = {"details": None, "code": status.canonical_code.value}
+
+    if status.description is not None:
+        status_json["message"] = status.description
+
+    return status_json
+
+
+def _extract_links(links: Sequence[trace_api.Link]):
+    """Convert span.links"""
+    if not links:
+        return None
+    extracted_links = []
+    for link in links:
+        trace_id = trace_api.format_trace_id(link.context.trace_id)[2:]
+        span_id = trace_api.format_span_id(link.context.span_id)[2:]
+        extracted_links.append(
+            {
+                "trace_id": trace_id,
+                "span_id": span_id,
+                "type": "CHILD_LINKED_SPAN",
+                "attributes": _extract_attributes(link.attributes),
+            }
+        )
+    return {"link": extracted_links}
+
+
+def _extract_events(events: Sequence[Event]):
+    """Convert span.events to dict."""
+    if not events:
+        return None
+    logs = []
+    for event in events:
+        logs.append(
+            {
+                "time": get_time_from_ns(event.timestamp),
+                "annotation": {
+                    "description": get_truncatable_str(event.name, 256),
```

Comment:
Why use 256 here, but MAX_LENGTH (which is 128) in other places?

Reply:
Looking at the protobuf file, the Annontation type allows strings of length 256 bytes. In general, strings have max length 128 (like for the name of the span).